apis/datadoghq/common/v1/agent_types.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package common
+
+import corev1 "k8s.io/api/core/v1"
+
+// AgentImageConfig defines the agent container image config.
+// +kubebuilder:object:generate=true
+type AgentImageConfig struct {
+	// Define the image to use:
+	// Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7.
+	// Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7.
+	// Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent.
+	// Use "agent" with the registry and tag configurations for <registry>/agent:<tag>.
+	// Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>.
+	Name string `json:"name,omitempty"`
```

Comment:
i'm a little confused by the comments; should we have a `Registry` field also?

Reply:
Registry is in `GlobalConfig`